java/src/main/java/org/digma/intellij/plugin/idea/psi/java/JavaSpanNavigationProvider.java

```diff
@@ -228,11 +232,14 @@ private void buildWithSpanAnnotation(VirtualFile virtualFile) {
             Query<PsiMethod> psiMethods = AnnotatedElementsSearch.searchPsiMethods(withSpanClass, GlobalSearchScope.fileScope(project, virtualFile));
             psiMethods = filterNonRelevantMethodsForSpanDiscovery(psiMethods);
             psiMethods.forEach(psiMethod -> {
-                SpanInfo spanInfo = SpanDiscoveryUtils.getSpanInfoFromWithSpanAnnotatedMethod(psiMethod);
-                if (spanInfo != null) {
-                    int offset = psiMethod.getTextOffset();
-                    var location = new SpanLocation(spanInfo.getContainingFileUri(), offset);
-                    spanLocations.put(spanInfo.getId(), location);
+                List<SpanInfo> spanInfos = JavaSpanDiscoveryUtils.getSpanInfoFromWithSpanAnnotatedMethod(psiMethod);
+                if (spanInfos != null) {
+                    spanInfos.forEach(spanInfo -> {
+                        Log.log(LOGGER::debug, "Found span info {} for method {}",spanInfo.getId(),spanInfo.getContainingMethod());
```

Comment:
now that method returns a list, 
1. maybe need to get rid of the null check?
2. the LOGGER::debug should be only once, for the method, isn't it?

Reply:
getSpanInfoFromWithSpanAnnotatedMethod can still return null.
I want a log message containing the span id too for every span we add to method info.  its can be changed to "Adding span info {} to method {}" if you want.